boom/legacy.py

```
@@ -171,24 +171,24 @@ def write_legacy_loader(selection=None, loader=BOOM_LOADER_GRUB1, cfg_path=None)
         chmod(path, BOOT_ENTRY_MODE)
-    except Exception as e:
-        _log_error("Error writing legacy configuration file %s: %s", path, e)
+    except (OSError, IOError) as err:
+        _log_error("Error writing legacy configuration file %s: %s", path, err)
         try:
             unlink(tmp_path)
-        except Exception:
-            _log_error("Error unlinking temporary path %s", tmp_path)
+        except (OSError, IOError) as err2:
+            _log_error("Error unlinking temporary path %s: %s", tmp_path, err2)
         raise e
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)